### ISSUE TITLE
Proper way to deal star fox 64 style overlays?

### ISSUE THREAD

**Issue by lizardman0** [N64Recomp/N64Recomp#77]
Hi, as mentioned on https://github.com/N64Recomp/N64Recomp/issues/66 sf64 has a bunch of overlays that map to the same vram address https://github.com/sonicdcer/sf64/blob/master/yamls/us/rev1/overlays.yaml 

It seems that the main problem is not even the overlapping addresses but the fact that there's no way to tell N64Recomp to  use dynamic `LOOKUP_FUNCTION` macros instead of fixed ones for targets inside overlays. I known that it's possible to specify relocatable symbols but that's not the same situation right? 

As a workaround, I made a quick and dirty script to replace(directly on the generated `.c` files) all the wrongly resolved functions with dynamic `LOOKUP_FUNCTION` and it seemed to work (game booted, and I even tested two levels).

In summary, does it make sense to add a way to specify overlay sections and force N64Recomp to output LOOKUP macros when the jump target is in an overlay section AND the current position is not in the same section?

**Comment by AngheloAlf** [comment]
N64Recomp emitting fixed function calls instead of `LOOKUP_FUNCTION` happens because the recompiler doesn't know how to handle those overlapping segments yet. It will be fixed when we add overlapping segments support.

**Comment by lizardman0** [comment]
Thanks for the quick answer! It makes sense

> For example if a jal is in section 3 and the one of the target functions is also in section 3, then the recompiler just has to emit the function call to that function in section 3 since it's unambiguous.

Nice, that's exactly what I did.  Only calls the "crossed" sections were replaced. The dump functionality hidden in the main really helped with this.

**Comment by Mr-Wiseguy** [comment]
Sorry it took so long for me to get to this, but I've pushed up a branch with a rewrite of the jal resolution logic that should fix this. Let me know if it works for you: https://github.com/N64Recomp/N64Recomp/tree/improved-jal-resolution

**Comment by Mr-Wiseguy** [comment]
I've tested these changes on a couple of ROMs with good results, so I've merged the change into the main branch and am closing the issue. Let me know if it doesn't solve it for your project and I'll reopen this.
